example_datascience/__main__.py

```diff
@@ -219,6 +436,104 @@ def main() -> int:
     evaluating_model(data=evaluate_data)
     serving_a_model(data=evaluate_data[0:10])
     monitor_the_model(data=evaluate_data)
+=======
+    preprocessor_pipeline, preprocessed_data = preprocessing_data(data)
+    training_model(preprocessor_pipeline, preprocessed_data, data["decision"], exploratory_model_analysis=False)
+    serving_a_model(data)
+
+    # # models = exp.compare_models(include=['lr', 'dt', 'knn', 'catboost'])
+    # models = exp.compare_models()
+    #
+    # # dt = exp.create_model('dt')
+    # # catboost = exp.create_model('catboost')
+    #
+    # # 3. Evaluating models
+    # # Experiment Tracking met mlflow (potentieel ook dataset tracking, maar voor nu doen we dat nog niet?)
+    #
+    # # save model
+    # # exp.save_model(catboost, 'catboost_pipeline')
+    # catboost = exp.load_model('catboost_pipeline')
+    #
+    # # 3.1 evaluating fairness-related metrics
+    # # y_pred = exp.predict_model(catboost, data=unseen_data).loc[:, "prediction_label"]
+    # y_pred = exp.predict_model(catboost, data=unseen_data)
+    # y_true = unseen_data.loc[:, "decision"]
+    # gender = unseen_data.loc[:, "gender"]
+    # # dp = demographic_parity_difference(y_true=y_true, y_pred=y_pred, sensitive_features=gender)
+    # # print(dp)
+    # # dp = demographic_parity_ratio(y_true=y_true, y_pred=y_pred, sensitive_features=gender)
+    # # Deze geeft een ratio van 0.67 aan wat betekent dat er vaker males worden geselecteerd dan females door het model
+    # # mfx = MetricFrame(metrics=accuracy_score, y_true=y_true, y_pred=y_pred, sensitive_features=gender)
+    # # sr = MetricFrame(metrics=selection_rate, y_true=y_true, y_pred=y_pred, sensitive_features=gender)
+    # # print(dp)
+    # # print(sr.by_group)
+    #
+    # # ## check fairness out of the box of pycaret
+    # # catboost_fairness = exp.check_fairness(catboost, sensitive_features=['gender'])
+    # # print(catboost_fairness)
+    #
+    # ## check mitigation of demographic disparity of gender via fairlearn
+    # # constraint = DemographicParity()
+    # # mitigator = ExponentiatedGradient(catboost, constraint)
```

Comment:
Could also be interesting to try out a postprocessing bias mitigation algorithm instead of only using pre-processing bias mitigation

Reply:
Yeah, I was thinking about this as well but I am doubting whether it would make this example project too large or not?